### ISSUE TITLE
JavaScript Error

### ISSUE THREAD

**Issue by UnaOmnia** [slowe/VirtualSky#48]
I am getting this error:

Uncaught ReferenceError: S is not defined  @ virtualsky.js?ver=5.0.3:3244

Which basically means the last line with the S. I will try to fix it but that is what I got out of the box.

**Comment by Gloupys** [comment]
Did you manage to make it work? I am having the same issue.

**Comment by slowe** [comment]
@UnaOmnia You'll have to give me a little more detail. Do you have an example somewhere of it not working? Are the [examples](http://slowe.github.io/VirtualSky/) working? Is the [embed version](http://slowe.github.io/VirtualSky/embed) working? Are you looking at a local copy? Which browser+version are you using?

**Comment by Gloupys** [comment]
@slowe Well not sure about him, but I am using the little tutorial here https://virtualsky.lco.global/#instructions with Chrome 71.0.3578.98 and the 0.7.0 version. I actually use it in a bigger project, as I am trying to add components on a customizable screen. I added the virtualsky.min.js script from the 0.7.0 folder into my project and copy pasted your HTML code, but I get an error regarding the S function, namely:

Uncaught ReferenceError: S is not defined
    at virtualsky.min.js:21

I didn't alter the given file I just followed the tutorial.

**Comment by slowe** [comment]
@Gloupys I have no control over the lco.global website you are using and it is at least a year or so since they updated it. The example on those old pages won't work with version 0.7.0 and that explains why you are getting issues. If you want to follow the old instructions you'll have to go back to a version before 0.7.0 (0.7.0 removed jQuery and uses a much smaller library I made as a replacement). If you'd like to use the latest version (0.7.0) you should follow the instructions on the up-to-date pages for this project at: http://slowe.github.io/VirtualSky/ 

Basically, you were getting `S is not defined` because you weren't including the stuQuery library.

**Comment by UnaOmnia** [comment]
dev.unaomnia.com

I got it working with an older version... I might take it from the top again before launch... 

http://dev.unaomnia.com/wp-content/themes/strappress-child/js/virtualsky/virtualsky.js?ver=5.0.3

I think it is 0.6.7 or something. You can see there. I had to go into one of the older versions. I did that because I had installed the project on a demo about 6 months back just fine. So I decided an older version should work just fine the same, and it did.

You are probably right about the stuQuery part... I did not include that... I wonder if it will interfere with including the regular jquery library... the new version is better for performance?

**Comment by Gloupys** [comment]
I did get stuquery instead and well the error seems gone eventhough I get an "$ is not defined" one now haha it seems to be an issue with Angular, though I dont' really get why

**Comment by slowe** [comment]
@Gloupys If you are using jQuery in a separate-to-this thing on your page you still need to include it otherwise you'll get a `$ is not defined` because `$` is the jQuery object. The console should probably tell you where the `$ is not defined` error is coming from.

@UnaOmnia As far as I'm aware my library (any version) does not interfere with jQuery. If you want to use an old version of this code, that is entirely up to you. The big change in version 0.7.0 was to slim down the initial load for the end-user from 337kB to 200kB by removing the dependency on jQuery. I also added the ability to customise the thickness of various lines and fixed the place search which was no longer working due to Yahoo deprecating their API.

As the Javascript error that this issue was opened for is not actually a bug in this project, I'm going to close this issue.

**Comment by UnaOmnia** [comment]
> I did get stuquery instead and well the error seems gone eventhough I get an "$ is not defined" one now haha it seems to be an issue with Angular, though I dont' really get why

https://api.jquery.com/jquery.noconflict/
https://stackoverflow.com/a/29697321/3759663
https://learn.jquery.com/using-jquery-core/avoid-conflicts-other-libraries/

That is probably the answer for him, and for me when I update to the latest version of Virtual Sky - with stuQuery.  Feel free to close but just thought that would help.

**Comment by slowe** [comment]
@UnaOmnia To be very clear: my library does not conflict with jQuery. However, if you remove the jQuery library, any separate-to-VirtualSky code on your page that requires jQuery will then produce a `$ is not defined` error. If you need jQuery for other reasons on the page you still need to include jQuery. For VirtualSky you *also* need to include stuQuery.

**Comment by TheBlackHacker** [comment]
Just include stuQuery before VirtualSky and it worked.
stuQuery: https://slowe.github.io/stuquery/
![image](https://user-images.githubusercontent.com/8444126/134661799-780e71c0-a15e-4243-b35c-0d7d817b7612.png)

**Comment by jumpjack** [comment]
Not: all examples at https://virtualsky.lco.global/#instructions are wrong because they  use "$.virtualsky" in place lof "S.virtualsky" (**stulibrary** is a custom replacement of jquery: S refers to stuquery, $ refers to jquery).

**Comment by slowe** [comment]
@jumpjack The page you link to is not under my control and refers to an older version of the code. If you want instructions for the current version of the code in this repository, you'd be best to go to https://slowe.github.io/VirtualSky/ That's the page linked to from this repository.
